sycl/source/detail/kernel_info.hpp

```diff
@@ -56,14 +56,63 @@ get_kernel_info(RT::PiKernel Kernel, const plugin &Plugin) {
 }
 
 // Device-specific methods
+template <typename Param>
+uint32_t get_kernel_device_specific_info_with_input(RT::PiKernel Kernel,
+                                                    RT::PiDevice Device,
+                                                    const void *In,
+                                                    size_t InSize,
+                                                    const plugin &Plugin) {
+  static_assert(is_kernel_device_specific_info_desc<Param>::value,
+                "Unexpected kernel_device_specific information descriptor");
+  static_assert(std::is_same<typename Param::return_type, uint32_t>::value,
+                "Unexpected return type");
+  static_assert(IsSubGroupInfo<Param>::value,
+                "Unexpected kernel_device_specific information descriptor for "
+                "query with input");
+  uint32_t Result;
+  // TODO catch an exception and put it to list of asynchronous exceptions
+  Plugin.call<PiApiKind::piKernelGetSubGroupInfo>(
+      Kernel, Device, PiInfoCode<Param>::value, InSize, In, sizeof(uint32_t),
+      &Result, nullptr);
+
+  return Result;
+}
+
+template <typename Param>
+uint32_t get_kernel_device_specific_info_with_input(RT::PiKernel Kernel,
+                                                    RT::PiDevice Device,
+                                                    sycl::range<3> In,
+                                                    const plugin &Plugin) {
+  size_t Input[3] = {In[0], In[1], In[2]};
+  return get_kernel_device_specific_info_with_input<Param>(
+      Kernel, Device, Input, 3 * sizeof(size_t), Plugin);
+}
+
 template <typename Param>
 typename std::enable_if<IsSubGroupInfo<Param>::value>::type
 get_kernel_device_specific_info_helper(RT::PiKernel Kernel, RT::PiDevice Device,
                                        const plugin &Plugin, void *Result,
                                        size_t Size) {
-  Plugin.call<PiApiKind::piKernelGetSubGroupInfo>(
-      Kernel, Device, PiInfoCode<Param>::value, 0, nullptr, Size, Result,
-      nullptr);
+  if constexpr (std::is_same_v<
+                    Param, info::kernel_device_specific::max_sub_group_size>) {
+    // max_sub_group_size requires additional information about intended work
+    // group size, so we use max work size from the device to avoid truncating
+    // the max sub-group size.
+    assert(Size >= sizeof(uint32_t) &&
+           "Unexpected result size for "
+           "info::kernel_device_specific::max_sub_group_size");
```

Comment:
This is a misleading assert message - one can confuse with the actual result of the query (because we're requesting a size).

Reply:
This assert has been removed.